python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -0,0 +1,750 @@
+		Gaffer.NodeAlgo.applyUserDefaults( self.__settingsNode )
+
+		searchFilter = _GafferSceneUI._RenderPassEditor.SearchFilter()
+		self.__filter = searchFilter
+
+		with mainColumn :
+
+			GafferUI.PlugLayout(
+				self.__settingsNode,
+				orientation = GafferUI.ListContainer.Orientation.Horizontal,
+				rootSection = "Settings"
+			)
+
+			_SearchFilterWidget( searchFilter )
+
+			self.__renderPassNameColumn = _GafferSceneUI._RenderPassEditor.RenderPassNameColumn()
+			self.__renderPassActiveColumn = _GafferSceneUI._RenderPassEditor.RenderPassActiveColumn()
+			self.__pathListing = GafferUI.PathListingWidget(
+				Gaffer.DictPath( {}, "/" ), # temp till we make an RenderPassPath
+				columns = [
+					self.__renderPassNameColumn,
+					self.__renderPassActiveColumn,
+				],
+				selectionMode = GafferUI.PathListingWidget.SelectionMode.Cells,
+				displayMode = GafferUI.PathListingWidget.DisplayMode.Tree,
+				horizontalScrollMode = GafferUI.ScrollMode.Automatic
+			)
+
+			self.__pathListing.buttonDoubleClickSignal().connectFront( Gaffer.WeakMethod( self.__buttonDoubleClick ), scoped = False )
+			self.__pathListing.keyPressSignal().connect( Gaffer.WeakMethod( self.__keyPress ), scoped = False )
+			self.__pathListing.buttonPressSignal().connectFront( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+
+			self.__settingsNode.plugSetSignal().connect( Gaffer.WeakMethod( self.__settingsPlugSet ), scoped = False )
+
+		self._updateFromSet()
+		self.__updateColumns()
+
+	__columnRegistry = collections.OrderedDict()
+
+	@classmethod
+	def registerOption( cls, groupKey, optionName, section = None, columnName = None ) :
+
+		if not columnName :
+			columnName = optionName.split( ":" )[-1]
+
+		GafferSceneUI.RenderPassEditor.registerColumn(
+			groupKey,
+			optionName,
+			lambda scene, editScope : _GafferSceneUI._RenderPassEditor._RenderPassEditorInspectorColumn(
+				GafferSceneUI.Private.OptionInspector( scene, editScope, optionName ),
+				columnName
+			),
+			section
+		)
+
+	# Registers a column in the Render Pass Editor.
+	# `inspectorFunction` is a callable object of the form
+	# `inspectorFunction( scene, editScope )` returning a
+	# `GafferSceneUI._RenderPassEditor._RenderPassEditorInspectorColumn` object.
+	@classmethod
+	def registerColumn( cls, groupKey, columnKey, inspectorFunction, section = None ) :
+
+		sections = cls.__columnRegistry.setdefault( groupKey, collections.OrderedDict() )
+		section = sections.setdefault( section, collections.OrderedDict() )
+
+		section[columnKey] = inspectorFunction
+
+	def __repr__( self ) :
+
+		return "GafferSceneUI.RenderPassEditor( scriptNode )"
+
+	def __firstValidScenePlug( self, node ):
+
+		for plug in GafferScene.ScenePlug.RecursiveOutputRange( node ) :
+			if not plug.getName().startswith( "__" ):
+				return plug
+		return None
+
+	def _updateFromSet( self ) :
+
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		# Decide what plug we're viewing.
+		self.__plug = None
```

Comment:
I know this is mirrored from the LightEditor, but I think we should be able to live without `self.__plug` - it's always the same as `__settingsNode["in"].getInput()`. That's what I did in the ImageInspector, and it went pretty well.

Reply:
`self.__plug` removed in 1140aa5